piano-hero/mobile/src/main/java/com/novoda/pianohero/PianoC4ToB5Activity.java

```diff
@@ -29,7 +28,10 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
     @Override
     protected void onResume() {
         super.onResume();
-        inputView.attach(new NotesPlayedDispatcher(presenter));
+        NotesPlayedDispatcher noteListener = new NotesPlayedDispatcher(presenter);
+        midiKeyboardDriver.attachListener(noteListener);
+        midiKeyboardDriver.open();
+        inputView.attach(noteListener);
```

Comment:
using the same noteListener for both - you could use the onscreen keyboard or a midi input device.

the usb driver didn't seem to work when I connected my piano to my pixelc. Dunno if it's for more raw input, like on AndroidThings, whereas on PixelC android handles (and consumes?) the usb midi events and you have to register via the midi service.